main.tf

```diff
@@ -24,18 +20,18 @@ resource "aws_sns_topic_subscription" "pipeline_updates" {
 }
 
 resource "aws_codestarnotifications_notification_rule" "pipeline_updates" {
-  count          = length(var.codepipelines)
+  for_each       = { for pipeline in var.codepipelines : pipeline.name => pipeline.arn }
   detail_type    = "FULL"
   event_type_ids = var.event_type_ids
-  name           = "slackNotification${var.codepipelines[count.index].name}"
-  resource       = var.codepipelines[count.index].arn
+  name           = join("-", [each.key, module.this.name])
```

Comment:
The reason why I have it in this order is we're using context.tf extensively in our modules and so the pipeline name already has the namespace, stage, etc; and then we just append the unique name for the notification.